machine-def.struct.js

```diff
@@ -33,6 +33,209 @@ module.exports = {
 
   },
+  
+  
+  // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
+  // FUTURE:
+  // In the machine specification, we always leverage "s-selection"; that is, the implementation exposes named
+  // inputs, allowing a caller invoking a machine to configure runtime arguments for particular inputs based
+  // on their meaning.  But this contrasts with how traditional subroutines in many programming languages are
+  // addressed.  Most languages use "c-selection" to assign runtime arguments to parameters (i.e. inputs) of
+  // a function.
+  //
+  // While these kind of syntactic/grammatical relations can be precisely imitated by runtime tooling, it is
+  // helpful to be able to declare it in a standardized way, as part of the machine definition itself.  This
+  // has practical utility for when full or partial serial assignment is the only option (e.g. a web-based API
+  // that must support taking in data via route/path parameters in its URL pattern; or a CLI script that must
+  // accept serial command-line arguments).
+  // > For an example, see:
+  // > • https://github.com/sailshq/machine-as-script/tree/e1a54d18d65809eb4cfa64bfe9db4ae925146e77#using-serial-command-line-arguments
+  // 
+  // ```
+  // args: ['someInput']
+  // ```
+  //
+  // I. What about variadic usage?
+  // Currently, the machine specification does not allow for variadic usage (although there is limited,
+  // non-standard support for this in machine-as-script).  This is something that makes perfect sense for
+  // future versions of the spec.  This would allow for numerous additional possibilities; e.g.
+  // ```
+  // args: [
+  //   ['foo', 'bar', 'baz'],             // << indicating order, and which inputs can be satisfied via serial usage (not all of them must be, necessarily -- imagine serial CLI arguments (`sails new foo`) vs. CLI options (`sails new foo --without='grunt'`) )
+  //   ['foo', 'baz'],                    // << indicating structual optional-ness of serial args (both of these usages existing indicates that either usage is acceptable)
+  //   ['foo', 'bar', 'baz', 'blazes[]'], // << n-ary/variadic functions that support bundling of extra serial arguments as an array, then passing that array in for a particular input (but note that this is the LAST serial argument, and also that it is only acceptable if it lists at least one more argument than every other serial usage that has been declared.  This is just to avoid ambiguity / unnecessary runtime performance degredation for extra data-type checking.)
+  //   ['done()'],                        // << this special, parentheses-trailed syntax indicates that this serial argument is actually a replacement for `exits` (note that this is not recommended with variadic usage, since the callback should always be the last argument, and variadic usage would force the callback to come first.  The name before the parentheses does not actually matter.)
```

Comment:
And this isn't as important anymore, now that await is widely used (and thanks to the extensive updates in [parley](https://npmjs.com/package/parley))